src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.Builder.cs

```diff
@@ -422,6 +477,50 @@ public bool Remove(T element)
                 return false;
             }
 
+            /// <summary>
+            /// Removes the specified element.
+            /// </summary>
+            /// <param name="item">The item to remove.</param>
+            /// <param name="equalityComparer">
+            /// The equality comparer to use in the search.
+            /// If <c>null</c>, <see cref="EqualityComparer{T}.Default"/> is used.
+            /// </param>
+            public void Remove(T item, IEqualityComparer<T>? equalityComparer)
+            {
+                int index = this.IndexOf(item, 0, _count, equalityComparer);
+
+                if (index >= 0)
+                {
+                    this.RemoveAt(index);
+                }
+            }
+
+            /// <summary>
+            /// Removes all the elements that match the conditions defined by the specified
+            /// predicate.
+            /// </summary>
+            /// <param name="match">
+            /// The <see cref="Predicate{T}"/> delegate that defines the conditions of the elements
+            /// to remove.
+            /// </param>
+            public void RemoveAll(Predicate<T> match)
+            {
+                List<int>? removeIndices = null;
+                for (int i = 0; i < _count; i++)
+                {
+                    if (match(_elements[i]))
+                    {
+                        removeIndices ??= new List<int>();
```

Comment:
Nit: it might be possible to do this allocation free, see the equivalent `List<T>` implementation:

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/List.cs#L879